src/psyclone/psyir/transformations/loop_fuse_trans.py

```diff
@@ -51,6 +54,8 @@ class LoopFuseTrans(LoopTrans):
     PSyIR of a Schedule after performing validity checks for the supplied
     Nodes. Examples are given in the descriptions of any children classes.
 
+    If loops have different named loop variables, the loop variable of the
+    second loop will be renamed if possible before fusion.
```

Comment:
Looking at the number of remaining TODOs I think it would be good to mention in the docstring that:
"Note that the validation of this transformation has still several shortcomings, specially for domain API loops, use at your own risk."

Reply:
Added